### PR TITLE
Hide into log the information that has been showing with show_error

### PR DESCRIPTION
**Pull request by rogeriopradoj**
This way, the common user in a production environment will NOT see messages like this:

"Unable to find template "form_escolher_unidade.html.twig" (looked into: /vagrant/application/themes/default, /vagrant/application/themes/default)."

### CODE REVIEW

**Comment by rogeriopradoj** [comment]
Hi, @edmundask .

I think it should be nice to put this change in next spark release, maybe 0.8.6.

I felt a little scared when I saw that message in a production environment.

Maybe you'd like to improve the messages I've put, but I think it's a nice start for you.

Thanks in advance!

**Comment by edmundask** [comment]
Thanks for the input, Rogerio. Your suggested code is somewhat incomplete. Were you just showing it as an example here? Because `$error_message` variable is not initialized or set anywhere. 

In any case, I get the idea. What I'm planning on doing is to add an extra config option (something like `environment_aware_errors`) where you could choose whether to react to error messages. For example, in the `production` environment there should be no errors visible to the user, so by setting `environment_aware_errors` to `TRUE` it would simply hide error messages and would log them instead. While in `development` environment those errors would be visible so that the developer could be aware of what is going on. 

On the other hand, setting `environment_aware_errors` to `FALSE` would display the errors regardless of the environment.

File: libraries/Twiggy.php [hunk 1]
```diff
@@ -60,8 +60,9 @@ public function __construct()
 		}
 		catch(Twig_Error_Loader $e)
 		{
-			log_message('error', 'Twiggy: failed to load the default theme');
-			show_error($e->getRawMessage());
+			$error_message = 'Twiggy: failed to load the default theme';
```
Comment (rogeriopradoj) [inline libraries/Twiggy.php:63]: It's been set here, for just this block.

It's just use the same messages that was already defined in code, and use that in both `log_message()` and `show_error()``

File: libraries/Twiggy.php [hunk 2]
```diff
@@ -311,8 +312,9 @@ public function theme($theme)
 	{
 		if(!is_dir(realpath($this->_themes_base_dir. $theme)))
 		{
-			log_message('error', 'Twiggy: requested theme '. $theme .' has not been loaded because it does not exist.');
-			show_error("Theme does not exist in {$this->_themes_base_dir}{$theme}.");
+			$error_message = 'Twiggy: requested theme '. $theme .' has not been loaded because it does not exist.';
```
Comment (rogeriopradoj) [inline libraries/Twiggy.php:315]: It's been set here, for just this block.

It's just use the same messages that was already defined in code, and use that in both `log_message()` and `show_error()``

File: libraries/Twiggy.php [hunk 3]
```diff
@@ -417,7 +419,9 @@ public function render($template = '')
 		}
 		catch(Twig_Error_Loader $e)
 		{
-			show_error($e->getRawMessage());
+			$error_message = 'Twiggy: failed to render';
```
Comment (rogeriopradoj) [inline libraries/Twiggy.php:422]: It's been set here, for just this block.

It's just use the same messages that was already defined in code, and use that in both `log_message()` and `show_error()``

File: libraries/Twiggy.php [hunk 4]
```diff
@@ -439,7 +443,9 @@ public function display($template = '')
 		}
 		catch(Twig_Error_Loader $e)
 		{
-			show_error($e->getRawMessage());
+			$error_message = 'Twiggy: failed to display';
```
Comment (rogeriopradoj) [inline libraries/Twiggy.php:446]: It's been set here, for just this block.

It's just use the same messages that was already defined in code, and use that in both `log_message()` and `show_error()``

**Comment by rogeriopradoj** [comment]
As you see, the `$error_message` is just set for one specific block. (Is it still incomplete in your opinion?)

I only create that for reuse the same message in both log_message and show_error, and the messages are the same that you have already put in code (except for the blocks in Twig::display() and Twig::render(), because they had only the raw messages from Twig Exception and not a Twiggy message from you).

And it's important to see now that the diff between the uses of `error_message` (let's say: tiny message) at `log_message` and `show_error` is that in the first one we put this tiny message followed by the Exception raw message, while in the latter we just show the tiny message.

I agree with your idea for create a extra config option. Anyway, I think all the errors should be put in log_message no matter how this config option is set.

I think that this pull request, at least until you build your idea, allow the current users of Twiggy (our fellow developers) use the last version with no bad surprises, like I had ;-)

Thanks

**Comment by edmundask** [comment]
I honestly didn't notice the `$error_message` in there, my bad. However, I want to keep error messages that come from Twig simply because they give insightful information. For example, when Twig fails to load the template file, it shows where it tried to look for that file. Of course, that's just one of the examples. So my point is, error messages should be as detailed as possible to make life easier for those who are developing applications with Twiggy.

**Pull request closed without merging**